sbi/neural_nets/estimators/categorical_net.py

```diff
@@ -21,108 +23,134 @@ class CategoricalNet(nn.Module):
 
     def __init__(
         self,
-        num_input: int,
-        num_categories: int,
-        num_hidden: int = 20,
-        num_layers: int = 2,
-        embedding_net: Optional[nn.Module] = None,
+        num_categories: Tensor,  # Tensor[int]
```

Comment:
why the comment? is it read by pyright? or can we explicitly use `Tensor[int]` as type hint?

Reply:
Can only typehint `Tensor` not `Tensor[int]`.